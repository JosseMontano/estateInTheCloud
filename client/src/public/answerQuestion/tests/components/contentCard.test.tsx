import ContentCard from "../../components/contentCard";
import { render, screen, fireEvent } from "@testing-library/react";

describe("contentCard", () => {
  const v = {
    id: 1,
    question: "You have lift",
  };
  const handleClick = vi.fn();

  beforeEach(() => {
    render(<ContentCard v={v} handleClick={handleClick} />);
  });

  test("should show the question", () => {
    /* expect("hi").toBe("hi"); */
    expect(screen.getByText("You have lift")).toBeDefined(); //toBeDefined is if exits
    expect(screen.getByDisplayValue)
  });

});
