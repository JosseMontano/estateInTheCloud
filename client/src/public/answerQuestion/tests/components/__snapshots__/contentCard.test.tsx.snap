// Vitest Snapshot v1

exports[`contentCard > the component should show be displayed 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-hLBbgP bDTcAK"
      >
        <h2
          class="sc-eDvSVe hqXXbc"
        >
          ?
        </h2>
        <h3
          class="sc-jSUZER hXOIrL"
        >
          Pregunta
        </h3>
        <p
          class="sc-gKPRtg cOqbwD"
        >
          You have lift
        </p>
        <button
          class="sc-iBYQkv gzwQFv"
        >
          Responder
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-hLBbgP bDTcAK"
    >
      <h2
        class="sc-eDvSVe hqXXbc"
      >
        ?
      </h2>
      <h3
        class="sc-jSUZER hXOIrL"
      >
        Pregunta
      </h3>
      <p
        class="sc-gKPRtg cOqbwD"
      >
        You have lift
      </p>
      <button
        class="sc-iBYQkv gzwQFv"
      >
        Responder
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
